src/main/java/de/tum/in/www1/artemis/repository/StudentParticipationRepository.java

```diff
@@ -47,6 +47,16 @@ public interface StudentParticipationRepository extends JpaRepository<StudentPar
             """)
     List<StudentParticipation> findByCourseIdWithEagerRatedResults(@Param("courseId") Long courseId);
 
+    @Query("""
+            SELECT DISTINCT p FROM StudentParticipation p
+            LEFT JOIN FETCH p.results r
+            LEFT JOIN p.team.students ts
+            WHERE p.exercise.course.id = :courseId
+                AND (p.student.id = :studentId OR ts.id = :studentId)
```

Comment:
Does this work? Just going by the name, "students" sounds like a list of multiple students. Can you just compare `ts.id = :studentId` to find one student in this list?

Reply:
```sql
select distinct /* omitted for brevity */
from participation studentpar0_
         left outer join result results1_ on studentpar0_.id = results1_.participation_id
         left outer join team team2_ on studentpar0_.team_id = team2_.id
         left outer join team_student students3_ on team2_.id = students3_.team_id
         left outer join jhi_user user4_ on students3_.student_id = user4_.id
         cross join exercise exercise5_
where studentpar0_.discriminator in ('SP', 'PESP')
  and studentpar0_.exercise_id = exercise5_.id
  and exercise5_.course_id = ?
  and (studentpar0_.student_id = ? or user4_.id = ?)
  and (results1_.rated is null or results1_.rated = 1)
  ```
 Locally it worked with a team exercise for me, Hibernate generates the SQL above for that method, does that makes sense to you?
 
My intention is getting the `StudentParticipation` if it is submitted by the student with `studentId` or submitted by a team whose member ids include `studentId`.